codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -1086,13 +1090,13 @@ buildNameLookupForBranchHash ::
   ([S.NamedRef C.Reference], [S.NamedRef C.Reference]) ->
   Transaction ()
 buildNameLookupForBranchHash mayExistingBranchIndex newBranchHash (newTermNames, removedTermNames) (newTypeNames, removedTypeNames) = do
-  newBranchHashId <- Q.saveBranchHash newBranchHash
+  newBranchHashId <- Q.expectBranchHashId newBranchHash
```

Comment:
Some functions needed to convert from BranchHash to BranchHashId, but there's a difference between always "saving" the branch hash to do it vs expecting one to exist.

In some cases we get the branch-hash from the API and shouldn't save it unless we know it's valid, so I added an `expectBranchHashId` for these situations.